src/time.jsx

```diff
@@ -0,0 +1,55 @@
+import React from 'react'
+import moment from 'moment'
+
+var Time = React.createClass({
+  displayName: 'Time',
+
+  propTypes: {
+    selected: React.PropTypes.string,
+    onTimeClick: React.PropTypes.func
+  },
+  getDefaultProps () {
+    return {
+      selected: '12:00'
+    }
+  },
+
+  handleTimeClick (time) {
+    console.log(time);
+    // let t = times[i];
+    this.props.onTimeClick(time);
+  },
+
+  renderTimes () {
+    let times = [];
+    let startOfDay = moment().startOf('day');
+    let endOfDay = moment().endOf('day');
+    let time = startOfDay;
+
+    while (time <= endOfDay) {
+      times.push(time.format('h:mm a'));
+      time = time.clone().add(30, 'minutes');
+    }
+
+    return (
+      <div>
+        {times.map(function(time, i){
+          return(
+            <div key={i} onClick={this.handleTimeClick.bind(this, time)}>{time}</div>
```

Comment:
It's probably better to use `time` instead of `key={i}`. Using the index as a key is an antipattern.

The binding looks OK to me. I'd probably write it like this though: `onClick={() => this.handleTimeClick(time)}`